middle_end/flambda/types/env/typing_env.rec.ml

```diff
@@ -854,6 +900,29 @@ let add_symbol_projection t var proj =
 let find_symbol_projection t var =
   Cached.find_symbol_projection (cached t) var
 
+let depth_variables t =
+  Cached.depth_variables (One_level.just_after_level t.current_level)
+
+let add_depth_variable t dv rec_info ~name_mode =
+  let level =
+    Typing_env_level.add_depth_variable (One_level.level t.current_level)
+      dv rec_info
+  in
+  let binding_time = t.next_binding_time in
+  let current_level =
+    let just_after_level =
+      Cached.add_depth_variable (cached t) dv rec_info binding_time name_mode
+    in
+    One_level.create (current_scope t) level ~just_after_level
+  in
+  with_current_level_and_next_binding_time t ~current_level
+    (Binding_time.succ t.next_binding_time)
+
+let mem_depth_variable ?min_name_mode t dv =
+  mem_by ?min_name_mode t dv
+    ~find:(fun dv t -> Depth_variable.Map.find dv (depth_variables t))
+    ~in_imported_names:(fun _ _ -> false)
```

Comment:
I'm not sure this line is correct.  I have a feeling that depth variables might be able to end up in exported typing environments.  The `Serializable` module below probably needs equipping to deal with them.  We should probably discuss this with @lpw25 .

Reply:
It depends what you mean by "in exported typing environments". They clearly end up in the terms of functions, which are in turn part of the types of functions. There are no free depth variables outside of functions though so, as long as depth variables bound by let are always simplified away, I don't think they can end up in the exported typing environment. Of course, you might not want to rely on the simplifying of let-bound ones, so it might be worth writing the code to handle them even if they can't appear.

Reply:
Seems like that would take a lot at the moment. Probably easier just to assert somewhere that there aren't free depth variables in exported types? Assuming we really can rely on simplifying them?

Reply:
That seems reasonable, especially for the alpha. Maybe leave a CR to add it later.